packages/drafts-realm/campaign-form.gts

```diff
@@ -0,0 +1,518 @@
+  currency: string = 'MYR',
+) => {
+  if (num === null || num === undefined) {
+    return '';
+  }
+
+  const formatter = new Intl.NumberFormat(locale, {
+    style: 'currency',
+    currency: currency,
+    minimumFractionDigits: 0, // No decimal places
+    maximumFractionDigits: 0, // No decimal places
+  });
+
+  let currentNumber = num;
+  if (typeof num === 'string') {
+    currentNumber = parseFloat(num);
+  }
+
+  return formatter.format(currentNumber as number);
+};
+
+const formatNumberWithSeparator = (
+  num: number | string | null | undefined,
+  isPercentage = false,
+) => {
+  if (num === null || num === undefined) {
+    return '';
+  }
+
+  let currentNumber = num;
+  if (typeof num === 'string') {
+    currentNumber = parseFloat(num);
+  }
+
+  return `${currentNumber.toLocaleString('en-US')}${isPercentage ? ' %' : ''}`;
+};
+
+class Isolated extends Component<typeof CampaignForm> {
+  <template>
+    <div class='campaign-form-isolated'>
+      <FieldContainer @label='Name' class='field'>
+        {{@model.name}}
+      </FieldContainer>
+      <FieldContainer @label='Status' class='field'>
+        {{@model.status}}
+      </FieldContainer>
+      <FieldContainer @label='Active' class='field'>
+        {{if @model.active 'Yes' 'No'}}
+      </FieldContainer>
+      <FieldContainer @label='Type' class='field'>
+        {{@model.type}}
+      </FieldContainer>
+      <FieldContainer @label='Parent Campaign' class='field'>
+        <@fields.parent_campaign />
+      </FieldContainer>
+      <FieldContainer @label='Description' class='field'>
+        {{@model.description}}
+      </FieldContainer>
+      <FieldContainer @label='Start Date' class='field'>
+        <@fields.start_date />
+      </FieldContainer>
+      <FieldContainer @label='End Date' class='field'>
+        <@fields.end_date />
+      </FieldContainer>
+      <FieldContainer @label='Num Sent in Campaign' class='field'>
+        {{formatNumberWithSeparator @model.number_sent}}
+      </FieldContainer>
+      <FieldContainer @label='Expected Response (%)' class='field'>
+        {{formatNumberWithSeparator @model.expected_response_percentage true}}
+      </FieldContainer>
+      <FieldContainer @label='Expected Revenue in Campaign' class='field'>
+        {{formatCurrency @model.expected_revenue}}
+      </FieldContainer>
+      <FieldContainer @label='Budgeted Cost in Campaign' class='field'>
+        {{formatCurrency @model.budgeted_cost}}
+      </FieldContainer>
+      <FieldContainer @label='Actual Cost in Campaign' class='field'>
+        {{formatCurrency @model.actual_cost}}
+      </FieldContainer>
+    </div>
+    <style>
+      .campaign-form-isolated {
+        display: grid;
+        gap: var(--boxel-sp-lg);
+        padding: var(--boxel-sp-xl);
+      }
+    </style>
+  </template>
+}
+
+class Embedded extends Component<typeof CampaignForm> {
+  <template>
+    {{@model.name}}
+  </template>
+}
+
+class Edit extends Component<typeof CampaignForm> {
```

Comment:
I think we have to rethink having a custom Edit component. I know previous PRs have been using it, but it seems that there are alot of data fields that we can make st we don't make this Edit component bloated. 

Date is a good example of one https://github.com/tintinthong/boxel/blob/main/packages/base/date.gts. Here they already have their own field class. 

Since this is mearly a form, you can just imagine a scenario where your campaign form doesn't have its own Edit component. But instead, it delegates to the Edit component that you specify. 

Hence, we should make all the new FieldDef like Percentage or Currency. I can pair with you on this

Reply:
If say, you have justification Edit component of this campaign for is needed. You can explicitly delegate it this [way](https://github.com/tintinthong/boxel/blob/3ea288e189fd7e2b69ac9cdba58377f3e5f49369/packages/drafts-realm/mortgage-calculator.gts#L313)